src/snowflake/snowpark/column.py

```diff
@@ -990,12 +994,14 @@ def _create_ast(
         ast = proto.Expr()
         if property is not None:
             prop_ast = getattr(ast, property)
+            fill_src_position(prop_ast.src)
             for attr, value in assign_fields.items():
                 setattr_if_not_none(prop_ast, attr, value)
             for attr, value in assign_opt_fields.items():
                 setattr_if_not_none(getattr(prop_ast, attr), "value", value)
             for attr, messg in copy_messages.items():
-                getattr(prop_ast, attr).CopyFrom(messg)
+                if messg is not None:
+                    getattr(prop_ast, attr).CopyFrom(messg)
```

Comment:
`CopyFrom` is a code smell in protobuf. I suspect that the idea of this helper is flawed – why not just create protobuf directly instead of calling this fn?

Reply:
This line is mainly utilized to copy the AST stored in existing column instances to the `col` field of the newly created operator AST.

Since the Column API allows arbitrary columns to be constructed independently of DataFrames or Sessions, we aren't guaranteed to have a top level AST (`Assign` or `Eval`) that we can always continue building down the chain of Column API calls, hence the current need to use `CopyFrom`.

This might be worth discussing and I have some preliminary discussion points in the original Column Expressions design doc I can share.

Reply:
Oh, I see. That's reasonable.

However, I still suspect that having this helper is counterproductive. Maybe let's discuss tomorrow in person?

Reply:
Follow up from in-person discussion with Arthur: This helper function complicates the logging of the AST at the cost of saving a few lines of code within each Column API. To make the logging logic easier to follow in each Column API it would be better to directly create and fill the 1:1 AST entity in each function itself at the cost of verbosity. (This kind of helper method would be better suited to the alternative of exposing a Builder interface to the client code to create/fill the AST as discussed previously.)